packages/web-components/fast-element/src/controller.ts

```diff
@@ -292,6 +320,83 @@ export class Controller extends PropertyChangeNotifier {
         return false;
     }
 
+    private finishInitialization() {
+        const element = this.element;
+        const boundObservables = this.boundObservables;
+
+        // If we have any observables that were bound, re-apply their values.
+        if (boundObservables !== null) {
+            const propertyNames = Object.keys(boundObservables);
+
+            for (let i = 0, ii = propertyNames.length; i < ii; ++i) {
+                const propertyName = propertyNames[i];
+                (element as any)[propertyName] = boundObservables[propertyName];
+            }
+
+            this.boundObservables = null;
+        }
+
+        const definition = this.definition;
+
+        // 1. Template overrides take top precedence.
+        if (this._template === null) {
+            if ((this.element as any).resolveTemplate) {
+                // 2. Allow for element instance overrides next.
+                this._template = (this.element as any).resolveTemplate();
+            } else if (definition.template) {
+                // 3. Default to the static definition.
+                this._template = definition.template || null;
+            }
+        }
+
+        // If we have a template after the above process, render it.
+        // If there's no template, then the element author has opted into
+        // custom rendering and they will managed the shadow root's content themselves.
+        if (this._template !== null) {
+            this.renderTemplate(this._template);
+        }
+
+        // 1. Styles overrides take top precedence.
+        if (this._styles === null) {
+            if ((this.element as any).resolveStyles) {
+                // 2. Allow for element instance overrides next.
+                this._styles = (this.element as any).resolveStyles();
+            } else if (definition.styles) {
+                // 3. Default to the static definition.
+                this._styles = definition.styles || null;
+            }
+        }
+
+        // If we have styles after the above process, add them.
+        if (this._styles !== null) {
+            this.addStyles(this._styles);
+        }
+
+        this.needsInitialization = false;
+    }
+
+    private renderTemplate(template: ElementViewTemplate | null | undefined) {
+        const element = this.element;
+        // When getting the host to render to, we start by looking
+        // up the shadow root. If there isn't one, then that means
+        // we're doing a Light DOM render to the element's direct children.
+        const host = getShadowRoot(element) || element;
+
+        if (this.view !== null) {
+            // If there's already a view, we need to unbind and remove through dispose.
+            this.view.dispose();
+            (this as Mutable<this>).view = null;
+        } else if (!this.needsInitialization) {
+            // If there was previous custom rendering, we need to clear out the host.
+            host.innerHTML = "";
```

Comment:
Might be better perf (see https://jsperf.com/innerhtml-vs-removechild/418):
```suggestion
            for (let child = host.firstChild; child !== null; child = host.firstChild) {
                host.removeChild(child);
            }
```

Reply:
Added a utility to do this and then refactored. Thanks!